src/systems/systems/src/hydraulic/linear_actuator.rs

```diff
@@ -799,15 +867,69 @@ impl<const N: usize> HydraulicLinearActuatorAssembly<N> {
         }
     }
 
-    fn update_lock_mechanism(&mut self, assembly_controller: &impl HydraulicAssemblyController) {
-        if assembly_controller.should_lock() {
-            self.rigid_body
-                .lock_at_position_normalized(assembly_controller.requested_lock_position())
-        } else {
+    fn update_hard_lock_mechanism(
+        &mut self,
+        assembly_controllers: &[impl HydraulicAssemblyController],
+    ) {
+        // The first controller requesting a lock locks the body
+        let mut no_lock = true;
+        for controller in assembly_controllers {
+            if controller.should_lock() {
+                self.rigid_body
+                    .lock_at_position_normalized(controller.requested_lock_position());
+
+                no_lock = false;
+
+                break;
+            }
+        }
+
+        if no_lock {
             self.rigid_body.unlock();
         }
     }
 
+    fn update_soft_lock_mechanism(&mut self, assembly_controllers: &[impl HydraulicLocking]) {
+        let mut no_lock = true;
+
+        let mut min_velocity = AngularVelocity::new::<radian_per_second>(-10000.);
+        let mut max_velocity = AngularVelocity::new::<radian_per_second>(10000.);
+
+        // Min velocity will be the max one amongst all the limit velocities requested
+        // Max velocity will be the min one amongst all the limit velocities requested
+        // This way we use the more restrictive speed limitation over all the controller demands
+        for (idx, controller) in assembly_controllers.iter().enumerate() {
```

Comment:
I am not sure this comment is quite accurate? It seems you `break` out of the for loop as soon as you find the first controller or actuator that should soft lock.

Reply:
Comment is accurate. But the break has nothing to do in here lol